test/Parse/pointer_conversion.swift.gyb

```diff
@@ -62,7 +62,9 @@ func mutablePointerArguments(_ p: UnsafeMutablePointer<Int>,
   takesMutableArrayPointer(&ii)
 
   // We don't allow these conversions outside of function arguments.
+  //expected-error @+1 {{use of extraneous '&'}}
   var x: UnsafeMutablePointer<Int> = &i // expected-error{{cannot pass immutable value as inout argument: 'i' is immutable}}
+  //expected-error @+1 {{use of extraneous '&'}}
```

Comment:
Why are there multiple diagnostics now which refer to the same thing? It seems like `preCheckExpression` might be a better place to diagnose this problem to avoid double diagnosing expressions...

Reply:
Well, this is technically two errors. One by parsing and one by the type checker. The third error is for the subsequent statement. I can certainly tweak parsing to not emit `InOutExpr` if it knows that it is will be diagnosed again later. I can see the argument for this on the grounds that if the `&` is truly extraneous, then it shouldn't generate trickle-down errors in Sema.

Reply:
I think there is no reason to type-check in this situation because we'd know that expression is invalid if diagnosing was done by `preCheckExpression`, multiple errors like this could be really confusing...

Reply:
I'll update the patch to hopefully improve diagnostic output quality.

Reply:
Thank you!